src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/ObjectWriter.cs

```diff
@@ -192,31 +199,55 @@ public void BuildSymbolDefinitionMap(ISymbolNode[] definedSymbols)
             }
         }
 
-        public void EmitSymbolDefinition(int currentOffset, NodeFactory factory)
+        private string GetSymbolToEmitForTargetPlatform(string symbol)
+        {
+            string symbolToEmit = symbol;
+
+            if (_targetPlatform.OperatingSystem == TargetOS.OSX)
+            {
+                // On OSX, we need to prefix an extra underscore to account for correct linkage of 
+                // extern "C" functions.
+                symbolToEmit = "_"+symbol;
```

Comment:
Is this only needed so that we can link with the bootstrapper code?

Bootstrapper is only temporary, so I would prefer temporary code to be in the temporary location.

Declaring the extern in the bootstrapper like this:

```c
extern void Foo(void) asm ("Foo");
```

Should let you import an extern function without the leading underscore.